staging/src/k8s.io/client-go/util/jsonpath/parser.go

```diff
@@ -447,3 +474,68 @@ func isAlphaNumeric(r rune) bool {
 func isBool(s string) bool {
 	return s == "true" || s == "false"
 }
+
+//UnquoteExtend is almost same as strconv.Unquote(), but it support parse single quotes as a string
+func UnquoteExtend(s string) (string, error) {
+	n := len(s)
+	if n < 2 {
+		return "", ErrSyntax
+	}
+	quote := s[0]
+	if quote != s[n-1] {
+		return "", ErrSyntax
+	}
+	s = s[1 : n-1]
+
+	if quote == '`' {
+		if contains(s, '`') {
+			return "", ErrSyntax
+		}
+		return s, nil
+	}
+	if quote != '"' && quote != '\'' {
+		return "", ErrSyntax
+	}
+	if contains(s, '\n') {
```

Comment:
newlines are valid inside json strings, so I'd imagine they would be valid for trying to match stuff with jsonpath as well